File: src/world.py
from typing import List, Optional
from blocktype import BlockType  # Import BlockType from blocktype.py

class World: 
   def __init__(self, width: int, height: int):
      self.width = width
      self.height = height
      self.grid: List [List[Optional[BlockType]]] = [] # 2D grid of Blocktype 
      self.generate_world()

    def generate_world(self):
        "Generate the world with blocks."
        for y in range(self.height):
            row = []
            for x in range(self.width):
                if y < self.height - 5:
                    row.append(BlockType.GRASS) # Top Layers are grass 
                    

File: src/world.ts
 // // src/world.ts
//export type BlockType = 'grass' | 'dirt' | 'stone';

//export class World {
    //private grid: BlockType[][];
    //public readonly width: number;
    //public readonly height: number;

    //constructor(width: number, height: number) {
        //this.width = width;
        //this.height = height;
        //this.grid = [];
        //this.generateWorld();
    //}

    //private generateWorld(): void {
        //for (let y = 0; y < this.height; y++) {
            //this.grid[y] = [];
            //for (let x = 0; x < this.width; x++) {
                //if (y < this.height - 5) {
                    //this.grid[y][x] = 'grass';
                //} else if (y < this.height - 1) {
                    //this.grid[y][x] = 'dirt';
                //} else {
                    //this.grid[y][x] = 'stone';
                //}
            //}
        //}
    //}

    public getBlock(x: number, y: number): BlockType | undefined {
        if (y >= 0 && y < this.height && x >= 0 && x < this.width) {
            return this.grid[y][x];
        }
        return undefined;
    }

    public setBlock(x: number, y: number, type: BlockType): void {
        if (y >= 0 && y < this.height && x >= 0 && x < this.width) {
            this.grid[y][x] = type;
        }
    }
}